No rc in repo rule bazel version

ittaiz opened the pull request:
fixes #86 
cc @damienmg @or-shachar

ittaiz commented:
oh no. Our lint check is broken :(

damienmg commented:
https://github.com/bazelbuild/bazel-integration-testing/blob/master/private/format.bzl#L57 is pointing to a rather old version of Bazel, Travis use the latest Bazel. Since the java_toolchain used for Bazel is defined in the Bazel repository this is probably just it.

ittaiz commented:
maybe. In any case please don't merge yet since these flags don't work with older versions.
Trying to understand what works with older versions. We might need to drop some of the really old versions.

ittaiz commented:
ok
@damienmg I needed to go through quite a lot of yak shaving for this 3 word change :(
fixed linting script by updating versions and shas
fixed create hash dict script since 0.17.0 was never released and I needed to introduce the "concept" of skipped versions
would appreciate your review again

ittaiz commented:
ok, I'm about to throw in the towel :( there are tens of issues due to the upgrade

ittaiz commented:
already with bazel 13 and buildifier 12 there are many issues 
I'm thinking of disabling the check and opening an issue for us to re-instate it wdyt?

damienmg commented on tools/bazel_hash_dict.bzl:
```diff
@@ -1,28 +1,20 @@
         "linux-x86_64": "13eae0f09565cf17fc1c9ce1053b9eac14c11e726a2215a79ebaf5bdbf435241",
     },
+    "0.16.0": {
+        "darwin-x86_64": "bb4720c027a991643be7dabee5c96a84c3776d6e2da92bf65df012a7cec15569",
```
Maybe not have 0.16.0 since there is already 0.16.1?

ittaiz commented:
Thanks!
Not sure I understand what you mean by fixing the test? I already ran
“./lint.sh fix” locally and pushed. Is this what you meant?
Also, re 0.16.0 this is the current logic of the create dict. I can
manually edit it.
Wdyt?
On Thu, 20 Sep 2018 at 22:14 Damien Martin-Guillerez <
notifications@github.com> wrote:

> *@damienmg* approved this pull request.
>
> The PR sounds good, just one comment. Disabling the check sounds good to
> keep that PR small but fixing the test seems like just running buildifier
> over it so should not be too hard (if it is not hiding skylint issues).
> ------------------------------
>
> In tools/bazel_hash_dict.bzl
> <https://github.com/bazelbuild/bazel-integration-testing/pull/87#discussion_r219285222>
> :
>
> >      "0.15.2": {
>          "darwin-x86_64": "84ee271274e7c8904a48f7894fe0569174d162c7428a19dda57fc6a6c95c628b",
>          "linux-x86_64": "13eae0f09565cf17fc1c9ce1053b9eac14c11e726a2215a79ebaf5bdbf435241",
>      },
> +    "0.16.0": {
> +        "darwin-x86_64": "bb4720c027a991643be7dabee5c96a84c3776d6e2da92bf65df012a7cec15569",
>
> Maybe not have 0.16.0 since there is already 0.16.1?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bazelbuild/bazel-integration-testing/pull/87#pullrequestreview-157425574>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABUIF7gW-wJ4IU6o2Xgsaoa7I0v5XGzqks5uc-kpgaJpZM4Ww9D1>
> .
>

damienmg commented:
I meant that normally running buildifier should have fix it but if not then
there is more to it.

Yes I meant manually edit it. I don't know why the update script added
0.16.0, it correctly passed 0.17.0...

On Thu, Sep 20, 2018, 10:06 PM Ittai Zeidman <notifications@github.com>
wrote:

> Thanks!
> Not sure I understand what you mean by fixing the test? I already ran
> “./lint.sh fix” locally and pushed. Is this what you meant?
> Also, re 0.16.0 this is the current logic of the create dict. I can
> manually edit it.
> Wdyt?
> On Thu, 20 Sep 2018 at 22:14 Damien Martin-Guillerez <
> notifications@github.com> wrote:
>
> > *@damienmg* approved this pull request.
> >
> > The PR sounds good, just one comment. Disabling the check sounds good to
> > keep that PR small but fixing the test seems like just running buildifier
> > over it so should not be too hard (if it is not hiding skylint issues).
> > ------------------------------
> >
> > In tools/bazel_hash_dict.bzl
> > <
> https://github.com/bazelbuild/bazel-integration-testing/pull/87#discussion_r219285222
> >
> > :
> >
> > > "0.15.2": {
> > "darwin-x86_64":
> "84ee271274e7c8904a48f7894fe0569174d162c7428a19dda57fc6a6c95c628b",
> > "linux-x86_64":
> "13eae0f09565cf17fc1c9ce1053b9eac14c11e726a2215a79ebaf5bdbf435241",
> > },
> > + "0.16.0": {
> > + "darwin-x86_64":
> "bb4720c027a991643be7dabee5c96a84c3776d6e2da92bf65df012a7cec15569",
> >
> > Maybe not have 0.16.0 since there is already 0.16.1?
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/bazelbuild/bazel-integration-testing/pull/87#pullrequestreview-157425574
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/ABUIF7gW-wJ4IU6o2Xgsaoa7I0v5XGzqks5uc-kpgaJpZM4Ww9D1
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bazelbuild/bazel-integration-testing/pull/87#issuecomment-423315304>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADjHf-MUw2KySoqwU0so7DwXV5bm1Xziks5uc_UsgaJpZM4Ww9D1>
> .
>

ittaiz commented:
0.16.0 exists.
0.17.0 doesn’t exist and i needed to update the script to ignore 0.17.0
(and get to 0.17.1).
Sounds good.
I’ll merge it as is and open an issue to revive the linting.
Honestly we use this library a lot internally but our biggest pains
currently are the fact that it can’t run on RBE and the infinite symlink
expansion.
These are serious issues we want to resolve.
On Thu, 20 Sep 2018 at 23:12 Damien Martin-Guillerez <
notifications@github.com> wrote:

> I meant that normally running buildifier should have fix it but if not then
> there is more to it.
>
> Yes I meant manually edit it. I don't know why the update script added
> 0.16.0, it correctly passed 0.17.0...
>
> On Thu, Sep 20, 2018, 10:06 PM Ittai Zeidman <notifications@github.com>
> wrote:
>
> > Thanks!
> > Not sure I understand what you mean by fixing the test? I already ran
> > “./lint.sh fix” locally and pushed. Is this what you meant?
> > Also, re 0.16.0 this is the current logic of the create dict. I can
> > manually edit it.
> > Wdyt?
> > On Thu, 20 Sep 2018 at 22:14 Damien Martin-Guillerez <
> > notifications@github.com> wrote:
> >
> > > *@damienmg* approved this pull request.
> > >
> > > The PR sounds good, just one comment. Disabling the check sounds good
> to
> > > keep that PR small but fixing the test seems like just running
> buildifier
> > > over it so should not be too hard (if it is not hiding skylint issues).
> > > ------------------------------
> > >
> > > In tools/bazel_hash_dict.bzl
> > > <
> >
> https://github.com/bazelbuild/bazel-integration-testing/pull/87#discussion_r219285222
> > >
> > > :
> > >
> > > > "0.15.2": {
> > > "darwin-x86_64":
> > "84ee271274e7c8904a48f7894fe0569174d162c7428a19dda57fc6a6c95c628b",
> > > "linux-x86_64":
> > "13eae0f09565cf17fc1c9ce1053b9eac14c11e726a2215a79ebaf5bdbf435241",
> > > },
> > > + "0.16.0": {
> > > + "darwin-x86_64":
> > "bb4720c027a991643be7dabee5c96a84c3776d6e2da92bf65df012a7cec15569",
> > >
> > > Maybe not have 0.16.0 since there is already 0.16.1?
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/bazelbuild/bazel-integration-testing/pull/87#pullrequestreview-157425574
> > >,
> > > or mute the thread
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ABUIF7gW-wJ4IU6o2Xgsaoa7I0v5XGzqks5uc-kpgaJpZM4Ww9D1
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/bazelbuild/bazel-integration-testing/pull/87#issuecomment-423315304
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/ADjHf-MUw2KySoqwU0so7DwXV5bm1Xziks5uc_UsgaJpZM4Ww9D1
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bazelbuild/bazel-integration-testing/pull/87#issuecomment-423317122>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABUIF-116L_hdD4x063gHH7vgwg7KF7Aks5uc_ajgaJpZM4Ww9D1>
> .
>

ittaiz commented:
please don't delete the branch just yet. We need it since we currently point to it and not to master (we'll change next week)
